docs/template-development.md

```diff
@@ -0,0 +1,85 @@
+
+Every template must depend on the [`@asyncapi/generator-react-sdk` package](https://github.com/asyncapi/generator-react-sdk), which contains a template file's basic components.
+
+## 2. `package.json` file
+
+Before the generation process runs, the generator installs the template into its dependencies. A `package.json` is necessary to identify the template name.
+
+The following block shows an example `package.json` file that points to the [React Render Engine](react-render-engine.md) and necessary dependencies:
+
+```json
+{
+  "name": "myTemplate",
+  "generator": {
+    "renderer": "react"
+  },
+  "dependencies": {
+    "@asyncapi/generator-react-sdk": "^0.2.25"
+  }
+}
+```
+
+## Template configuration
+
+You must configure the generator's `package.json` file to contain JSON objects with the required parameters for template configuration, such as:
+
+|Name|Type|Description|
+|---|---|---|
+|`renderer`| String | Its value can be either `react` or `nunjucks` (default).
+|`supportedProtocols`| [String] | A list with all the protocols this template supports.
+|`parameters`| Object[String, Object] | An object with all the parameters that can be passed when generating the template. When using the command line, it's done by indicating `--param name=value` or `-p name=value`.
+|`parameters[param].description`| String | A user-friendly description about the parameter.
+|`parameters[param].default`| Any | Default value of the parameter if not specified. Shouldn't be used for mandatory `required=true` parameters.
+|`parameters[param].required`| Boolean | Whether the parameter is required or not.
+
+These are predefined configurations which help generator achieve specific set of tasks throughout the generation process. The `generator` property from 'package.json' contains all the configuration information. To learn more about template configuration and various supported parameters. Read more about the [configuration file](configuration-file.md)
+
+> Whenever you make a change to the package.json, make sure you perform an update by running `npm install`;  this command synchronizes with the package-lock.json and validates the file.
+
+## Render engines
+
+Render engines generate code, documentation, markdown, diagrams, or anything else you may have specified to be generated as output. AsyncAPI templates support multiple render engines, including [Nunjucks](nunjucks-render-engine.md) and [React](react-render-engine.md). 
+
+Each rendering engine has its process and unique features, affecting how you write templates. While you are free to use the render engine of your choice, we recommend using the React render engine for your projects.
+
+> The [AsyncAPI React SDK](https://github.com/asyncapi/generator-react-sdk) is a set of components and functions that use React as a render engine in the generator.
+## Hooks
```

Comment:
we could add a real example here, like use case that someone wants to do changes in AsyncAPI document before it goes into generation

```
module.exports = {
  'generate:before': function switchOperations(generator) {
      const newVersion = generator.templateParams.version;
      if (!newVersion) return;
      
      const asyncapi = JSON.parse(generator.originalAsyncAPI);
      asyncapi.info.version = newVersion;
      generator.originalAsyncAPI = JSON.stringify(asyncapi);
  };
};
```

please check locally if above code actually works 😄 tbh I'm not sure 😅 

and you explain that this is a different way of supporting `version` parameter that was explained in previous section. That this way is better

then template looks like
```
<Text>App name is: **{ asyncapi.info().title() }**</Text>
<Text>Version is: **{asyncapi.info.version()}**</Text>
```

and important is that you override the version in the Template Context. So you can for example still use some reusable hooks like:
- https://github.com/asyncapi/nodejs-template/blob/master/package.json#L36
- https://github.com/asyncapi/nodejs-template/blob/master/package.json#L110

and if someone will do `-p version=2.0.0 asyncapiFileDir="./"` then not only markdown doc will be generated but also the asyncapi file that was passed to generator, but with overwritten version. 

I hope that makes sense 😅 lemme know if more explanation is needed